src/api/profile/ElevationProfile.class.js

```diff
@@ -1,67 +1,32 @@
-import { API_SERVICE_ALTI_BASE_URL } from '@/config'
-import { CoordinateSystems } from '@/utils/coordinateUtils'
-import log from '@/utils/logging'
-import axios from 'axios'
 import { LineString } from 'ol/geom'
 
-/**
- * @typedef GeoJSON A GeoJSON object
- * @property {string} type
- * @property {[[Number]]} coordinates
- */
-
-export class ElevationProfilePoint {
-    /**
-     * @param {Number} dist Distance from first to current point
-     * @param {[Number, Number]} coordinate Coordinate of this point in EPSG:3857
-     * @param {Number} elevation In the COMB elevation model
-     */
-    constructor(dist, coordinate, elevation) {
-        this._dist = dist
-        this._coordinate = coordinate
-        this._elevation = elevation
-    }
-
-    get dist() {
-        return this._dist
-    }
-
-    get coordinate() {
-        return this._coordinate
-    }
-
-    /** @returns {Number} The COMB elevation */
-    get elevation() {
-        return this._elevation
-    }
-
-    // no setter
-}
-
 /**
  * Representation of data coming from `service-alti`'s backend. Also encapsulate most business
  * calculation related to profile (hiking time, slop/distance, etc...)
  */
-export class ElevationProfile {
-    /** @param {ElevationProfilePoint[]} points */
-    constructor(points) {
-        /** @type {ElevationProfilePoint[]} */
-        this.points = [...points]
+export default class ElevationProfile {
+    /** @param {ElevationProfileSegment[]} segments */
+    constructor(segments) {
+        /** @type {ElevationProfileSegment[]} */
+        this.segments = [...segments]
+    }
+
+    get points() {
+        return this.segments.flatMap((segment) => segment.points)
     }
 
     /** @returns {Number} */
     get length() {
         return this.points.length
     }
 
-    /** @returns {Boolean} True if the profile has at least 2 points */
     get hasData() {
-        return this.length >= 2
+        return !!this.segments.find((segment) => segment.hasData)
     }
 
     /** @returns {Number} */
     get maxDist() {
-        return (this.hasData && this.points[this.length - 1].dist) || 0
+        return this.segments.slice(-1)[0].points.slice(-1)[0].dist
```

Comment:
What happen here if we have no segments ?

Reply:
then it will throw an error 🙈